runtime/src/main/java/org/corfudb/runtime/view/stream/BackpointerStreamView.java

```diff
@@ -286,23 +288,32 @@ protected boolean followBackpointers(final UUID streamId,
             // Now calculate the next address
             // Try using backpointers first
 
-            log.trace("followBackPointers: calculate the next address");
+            log.trace("followBackpointers: calculate the next address");
 
             if (!runtime.getParameters().isBackpointersDisabled() && d.hasBackpointer(streamId)) {
                 long tmp = d.getBackpointer(streamId);
-                log.trace("followBackPointers: backpointer points to {}", tmp);
+                log.trace("followBackpointers: backpointer points to {}", tmp);
                 // if backpointer is a valid log address or Address.NON_EXIST
                 // (beginning of the stream), do not single step back on the log
                 if (Address.isAddress(tmp) || tmp == Address.NON_EXIST) {
                     currentAddress = tmp;
                     singleStep = false;
+                    if (!startingSingleStep) {
+                        // A started single step period finishes here, refresh flag for next cycle.
+                        log.info("followBackpointers[{}]: Found backpointer for this stream at address {}. Stop single step downgrade.");
```

Comment:
In a good case. This method will start with `startingSingleStep` set to false.
Will we log this statement every time this method is run? Won't that cause a lot of logging?

Reply:
@zalokhan It is set to true at the beginning, initially, it will only be set to false once the singleStep is actually started, which it is no longer 'starting' cause it started already.